trino/client.py

```diff
@@ -758,12 +677,18 @@ def execute(self, additional_http_headers=None) -> TrinoResult:
         self._warnings = getattr(status, "warnings", [])
         if status.next_uri is None:
             self._finished = True
-        self._result = TrinoResult(self, status.rows, self._experimental_python_types)
+
+        rows = self._row_mapper.map(status.rows) if self._row_mapper else status.rows
+
+        self._result = TrinoResult(self, rows)
         return self._result
 
     def _update_state(self, status):
         self._stats.update(status.stats)
         self._update_type = status.update_type
+        if not self._row_mapper:
```

Comment:
I think it's possible that the columns are not yet known at this point. The row mapper would not be recreated in case the columns arrive later.

Reply:
That is correct, but then the `RowMapperFactory` will just return `None`